app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -102,4 +107,103 @@ def has_required_host?
   def self.calculate_power_state(raw_power_state)
     POWER_STATES[raw_power_state] || super
   end
+
+  #
+  # This method is executed in the UI worker.
+  # It adds a task to calculate the URL of oVirt UI for this virtual machine.
+  #
+  # @return [Integer] The identifier of the task.
+  #
+  def queue_generate_ui_url
+    # Create the task options:
+    task_options = {
+      action: "Generate oVirt UI URL for VM '#{name}'"
+    }
+
+    # Create the queue options:
+    queue_options = {
+      class_name: self.class.name,
+      instance_id: id,
+      method_name: 'generate_ui_url',
+      priority: MiqQueue::HIGH_PRIORITY,
+      role: 'ems_operations',
+      zone: my_zone,
+      args: []
+    }
```

Comment:
I think that the style checkers (Rubocop, Code climate, etc) will tell you that they don't like this modern and nice syntax for hashes. You will have the older and not so nice syntax:

```ruby
:class_name => self.class.name,
:instance_id: => id,
...
```

They will also probably insist to have the `=>` aligned.